### PR TITLE
Fix Multi-function declaration in Python domain has cramped vertical spacing   in latexpdf output (#9940)

### LINKED ISSUE #9940: Multi-function declaration in Python domain has cramped vertical spacing in latexpdf output
### How to Reproduce

```
.. py:function:: __dpd_addsd3 (a, b)
                 __bid_addsd3 (a, b)
                 __dpd_adddd3 (a, b)
                 __bid_adddd3 (a, b)
                 __dpd_addtd3 (a, b)
                 __bid_addtd3 (a, b)

    something
```

### Additional context

Mentioned at https://github.com/sphinx-doc/sphinx/issues/9924#issuecomment-984837203

### PR DESCRIPTION
**Pull request by jfbu**
Avoid `\pysiglinewithargsret` tricks to get correct vertical spacing
in multiline declarations in case mark-up inserts a `\phantomsection`.

Activate such tricks also in `\pysigline`, but now with the
`\phantomsection` test.

Fix #9940

Before:

Capture d’écran 2021-12-05 à 15 55 22

After:

Capture d’écran 2021-12-05 à 15 34 00

Relates #7241, #8980, #8995, https://github.com/sphinx-doc/sphinx/issues/9924#issuecomment-984837203

### CODE REVIEW

**Comment by jfbu** [comment]
Rebased onto 4.x

File: CHANGES [hunk 1]
```diff
@@ -39,6 +39,8 @@ Bugs fixed
 * #9925: LaTeX: prohibit also with ``'xelatex'`` line splitting at dashes of
   inline and parsed literals
 * #9944: LaTeX: extra vertical whitespace for some nested declarations
+* #9940: Multi-function declaration in Python domain has cramped vertical
```
Comment (tk0miya) [inline CHANGES]: Please add a category:
```suggestion
* #9940: LaTeX: Multi-function declaration in Python domain has cramped vertical
```

**Review by tk0miya — APPROVED** [review]
LGTM with nits.

File: CHANGES (hunk 1 above)
Comment (jfbu) [reply CHANGES]: category added!

**Comment by tk0miya** [comment]
👍

**Review by tk0miya — APPROVED** [review]

**Pull request merged**